File: renderer-mock/create-test-images.py
#!/usr/bin/env python3
"""
Create placeholder background images for testing the mock renderer.
These are simple colored rectangles that match the required dimensions.
"""

from PIL import Image, ImageDraw, ImageFont
import os

# Create assets/backgrounds directory
os.makedirs('assets/backgrounds', exist_ok=True)

# Image specifications
WIDTH = 3375   # 11.25 inches at 300 DPI (11" + 0.125" bleed each side)
HEIGHT = 2625  # 8.75 inches at 300 DPI (8.25" + 0.25" bleed top/bottom)

# Page descriptions for The Adventure Compass story
pages = [
    ("page01_bedroom", "Bedroom Scene", (135, 206, 235)),  # Sky blue
    ("page02_bedroom_night", "Bedroom Night", (25, 25, 112)),  # Midnight blue
    ("page03_forest", "Forest Path", (34, 139, 34)),  # Forest green
    ("page04_mountain", "Mountain View", (139, 137, 137)),  # Gray
    ("page05_sky", "Sky Scene", (135, 206, 235)),  # Sky blue
    ("page06_sea", "Ocean Scene", (0, 191, 255)),  # Deep sky blue
    ("page07_picnic", "Picnic Area", (255, 228, 196)),  # Bisque
    ("page08_cave", "Cave Entrance", (105, 105, 105)),  # Dim gray
    ("page09_garden", "Garden Scene", (144, 238, 144)),  # Light green
    ("page10_town", "Town Square", (255, 218, 185)),  # Peach puff
    ("page11_bedroom_return", "Bedroom Return", (135, 206, 235)),  # Sky blue
    ("page12_compass_glow", "Compass Glow", (255, 215, 0)),  # Gold
    ("page13_keepsake_frame", "Keepsake Frame", (255, 228, 196)),  # Bisque
    ("page14_dedication_frame", "Dedication Frame", (255, 228, 196))  # Bisque
]

print("🎨 Creating placeholder background images...")

for filename, description, color in pages:
    # Create image with solid color background
    img = Image.new('RGB', (WIDTH, HEIGHT), color)
    draw = ImageDraw.Draw(img)
    
    # Add page number and description
    try:
        # Try to use a larger font
        font = ImageFont.truetype("/System/Library/Fonts/Arial.ttf", 120)
    except:
        try:
            # Fallback to default font
            font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", 120)
        except:
            # Use default font
            font = ImageFont.load_default()
    
    # Draw page info
    text = f"{filename}\n{description}\n11.25\" × 8.75\" @ 300 DPI\n(11\" × 8.25\" trim)"
    
    # Calculate text position (centered)
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    
    x = (WIDTH - text_width) // 2
    y = (HEIGHT - text_height) // 2
    
    # Draw text with outline for visibility
    draw.text((x-2, y-2), text, fill=(0, 0, 0), font=font)  # Black outline
    draw.text((x+2, y+2), text, fill=(0, 0, 0), font=font)  # Black outline
    draw.text((x, y), text, fill=(255, 255, 255), font=font)  # White text
    
    # Save image
    filepath = f"assets/backgrounds/{filename}.png"
    img.save(filepath, "PNG", dpi=(300, 300))
    print(f"  ✅ Created {filepath}")

print(f"🎉 Created {len(pages)} placeholder background images!")
print("📁 Images saved to: assets/backgrounds/")
print("🔍 Dimensions: 3375×2625 pixels (11.25×8.75 inches @ 300 DPI)")
print("📏 Trim size: 11×8.25 inches (with 0.125\" bleed each side, 0.25\" top/bottom)")
print("🎨 Colors: Each page has a unique color representing the scene")
print("")
print("Next steps:")
print("1. Run: node server.js")
print("2. Test: curl http://localhost:8787/health")
print("3. Test render: curl -X POST http://localhost:8787/render -H 'Content-Type: application/json' -d '{\"orderId\":\"TEST-001\",\"pages\":[{\"background\":\"http://localhost:8787/assets/backgrounds/page01_bedroom.png\"}],\"cover\":{\"front_background\":\"http://localhost:8787/assets/backgrounds/page01_bedroom.png\",\"title\":\"Test Book\"}}'")

File: renderer-mock/create-test-images.js
const fs = require('fs');
const path = require('path');

// Create assets/backgrounds directory
const assetsDir = path.join(__dirname, 'assets', 'backgrounds');
if (!fs.existsSync(assetsDir)) {
    fs.mkdirSync(assetsDir, { recursive: true });
}

// Image specifications
const WIDTH = 3375;   // 11.25 inches at 300 DPI (11" + 0.125" bleed each side)
const HEIGHT = 2625;  // 8.75 inches at 300 DPI (8.25" + 0.25" bleed top/bottom)

// Page descriptions for The Adventure Compass story
const pages = [
    { filename: "page01_bedroom", description: "Bedroom Scene", color: [135, 206, 235] }, // Sky blue
    { filename: "page02_bedroom_night", description: "Bedroom Night", color: [25, 25, 112] }, // Midnight blue
    { filename: "page03_forest", description: "Forest Path", color: [34, 139, 34] }, // Forest green
    { filename: "page04_mountain", description: "Mountain View", color: [139, 137, 137] }, // Gray
    { filename: "page05_sky", description: "Sky Scene", color: [135, 206, 235] }, // Sky blue
    { filename: "page06_sea", description: "Ocean Scene", color: [0, 191, 255] }, // Deep sky blue
    { filename: "page07_picnic", description: "Picnic Area", color: [255, 228, 196] }, // Bisque
    { filename: "page08_cave", description: "Cave Entrance", color: [105, 105, 105] }, // Dim gray
    { filename: "page09_garden", description: "Garden Scene", color: [144, 238, 144] }, // Light green
    { filename: "page10_town", description: "Town Square", color: [255, 218, 185] }, // Peach puff
    { filename: "page11_bedroom_return", description: "Bedroom Return", color: [135, 206, 235] }, // Sky blue
    { filename: "page12_compass_glow", description: "Compass Glow", color: [255, 215, 0] }, // Gold
    { filename: "page13_keepsake_frame", description: "Keepsake Frame", color: [255, 228, 196] }, // Bisque
    { filename: "page14_dedication_frame", description: "Dedication Frame", color: [255, 228, 196] } // Bisque
];

console.log("🎨 Creating placeholder background images...");

// Create a simple HTML file that will generate the images
const htmlContent = `
<!DOCTYPE html>
<html>
<head>
    <title>Generate Test Images</title>
</head>
<body>
    <canvas id="canvas" width="${WIDTH}" height="${HEIGHT}" style="border: 1px solid #ccc;"></canvas>
    <div id="downloads"></div>
    
    <script>
        const canvas = document.getElementById('canvas');
        const ctx = canvas.getContext('2d');
        const downloads = document.getElementById('downloads');
        
        const pages = ${JSON.stringify(pages)};
        
        pages.forEach((page, index) => {
            // Clear canvas
            ctx.clearRect(0, 0, ${WIDTH}, ${HEIGHT});
            
            // Fill with background color
            ctx.fillStyle = \`rgb(\${page.color[0]}, \${page.color[1]}, \${page.color[2]})\`;
            ctx.fillRect(0, 0, ${WIDTH}, ${HEIGHT});
            
            // Add text
            ctx.fillStyle = 'white';
            ctx.font = 'bold 120px Arial';
            ctx.textAlign = 'center';
            ctx.textBaseline = 'middle';
            
            const text = \`\${page.filename}\\n\${page.description}\\n11.25" × 8.75" @ 300 DPI\\n(11" × 8.25" trim)\`;
            const lines = text.split('\\n');
            const lineHeight = 140;
            const startY = ${HEIGHT/2} - (lines.length * lineHeight) / 2;
            
            lines.forEach((line, i) => {
                ctx.fillText(line, ${WIDTH/2}, startY + (i * lineHeight));
            });
            
            // Convert to blob and download
            canvas.toBlob((blob) => {
                const url = URL.createObjectURL(blob);
                const a = document.createElement('a');
                a.href = url;
                a.download = \`\${page.filename}.png\`;
                a.textContent = \`Download \${page.filename}.png\`;
                a.style.display = 'block';
                a.style.margin = '10px';
                downloads.appendChild(a);
                
                // Auto-click after a short delay
                setTimeout(() => a.click(), index * 100);
            }, 'image/png');
        });
        
        console.log('All images generated! Check the downloads folder.');
    </script>
</body>
</html>
`;

// Write the HTML file
fs.writeFileSync(path.join(__dirname, 'generate-images.html'), htmlContent);

console.log(`🎉 Created HTML generator: generate-images.html`);
console.log(`📁 Open this file in your browser to generate the test images`);
console.log(`🔍 Dimensions: ${WIDTH}×${HEIGHT} pixels (11.25×8.75 inches @ 300 DPI)`);
console.log(`📏 Trim size: 11×8.25 inches (with 0.125" bleed each side, 0.25" top/bottom)`);
console.log(`🎨 Colors: Each page has a unique color representing the scene`);
console.log("");
console.log("Next steps:");
console.log("1. Open generate-images.html in your browser");
console.log("2. Images will auto-download to your Downloads folder");
console.log("3. Move the downloaded PNG files to assets/backgrounds/");
console.log("4. Run: node server.js");
console.log("5. Test: curl http://localhost:8787/health");
